src/OpenTracing.Contrib.NetCore/AspNetCore/HostingEventProcessor.cs

```diff
@@ -92,6 +94,20 @@ public bool ProcessEvent(string eventName, object arg)
             }
         }
 
+        private static string GetDisplayUrl(HttpRequest request)
+        {
+            if(request.Host.HasValue)
+                return request.GetDisplayUrl();
+
+            // HTTP 1.0 requests are not required to provide a Host to be valid
+            // Since this is just for display, we can provide a string that is
+            // not an actual Uri with only the fields that are specified.
+            // request.GetDisplayUrl(), used above, will throw an exception
+            // if request.Host is null.
+            var host = request.Host.HasValue ? request.Host.Value : NoHostSpecified;
```

Comment:
You already have the if-check at the beginning of the method so there's no need for another check here.